sound/soc/sof/ipc4-pcm.c

```diff
@@ -89,80 +222,74 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	if (!trigger_list)
 		return -ENOMEM;
 
+	mutex_lock(&ipc4_data->trigger_mutex);
+
 	/*
 	 * IPC4 requires pipelines to be triggered in order starting at the sink and
-	 * walking all the way to the source. So traverse the pipeline_list in the reverse order.
-	 * Skip the pipelines that have their skip_during_fe_trigger flag set or if they're already
-	 * in the requested state. If there is a fork in the pipeline, the order of triggering
-	 * between the left/right paths will be indeterministic. But the sink->source trigger order
-	 * sink->source would still be guaranteed for each fork independently.
+	 * walking all the way to the source. So traverse the pipeline_list in the order
```

Comment:
The existing first sentence is incorrect and states the in IPC4 the order is always sink->source, which is no longer true and the rest of the comment explains the different directions used.

Can be a fixup.

Reply:
what makes you think the order is not sink>source @ujfalusi ?

It's written everywhere in all the documentation I've seen.

Reply:
@plbossart what he means is its sink->source during start and the reverse during pause/stop. I'll fix this